vtm.js

```diff
@@ -190,28 +252,103 @@ const app = Vue.createApp({
           },
         ],
       },
+      disciplines:{
+        id: "Disciplines",
+        resource: [0,1,1,0,0,0],
+        data: [
+          {
+            id: 'Animalism',
+            value: 0,
+          },
+          {
+            id: 'Auspex',
+            value: 0,
+          },
+          {
+            id: 'Blood Sorcery',
+            value: 0,
+          },
+          {
+            id: 'Celerity',
+            value: 0,
+          },
+          {
+            id: 'Dominate',
+            value: 0,
+          },
+          {
+            id: 'Fortitude',
+            value: 0,
+          },
+          {
+            id: 'Obfuscate',
+            value: 0,
+          },
+          {
+            id: 'Potence',
+            value: 0,
+          },
+          {
+            id: 'Presence',
+            value: 0,
+          },
+          {
+            id: 'Protean',
+            value: 0,
+          },
+        ],
+      },
     };
   },
+  
   template: `
-  <stat-section 
+  <character-info
+    :bio="biography"
+    :clans="clans">
+  </character-info>
+  <attribute-section 
     :stats="attributes"
-    @stat-section-change="$event[0].value=$event[1]"
-    >
-  </stat-section>
-  <stat-section 
+    @stat-section-change="$event[0].value=$event[1]">
+  </attribute-section>
+  <skill-section 
     :stats="skills"
-    @stat-section-change="$event[0].value=$event[1]"
-    >
-  </stat-section>`
+    :distributions="skillDistributions"
+    @stat-section-change="$event[0].value=$event[1]">
+  </skill-section>
+  <discipline-section 
+    v-if="biography.clan"
+    :stats="disciplines"
+    :clan="biography.clan"
+    @stat-section-change="$event[0].value=$event[1]">
+  </discipline-section>`
 });
 
-/** Displays entire section of attributes or skills
- * receives events from child component and check if change is possible in resources
- * if changes are possible, emit event to top component to make changes, otherwise don't
- */
-app.component('stat-section', {
+const statSectionMixin = {
   props: ['stats'],
   emits: ['statSectionChange'],
+  methods: {
+    /** 
+      * if we are adding
+      * checks if change is allowed
+      * if change is not allowed, we will try lower value
+      * first allowed value is emitted
+      * if we are not adding, no restriction check is needed, so we emit event
+      */
+    emitAllowedChange(received_event) {
+      var i = received_event[1];    
```

Comment:
hovoril som si, ze received_event je mozno zla konvencia, ze miesto toho by mala byt ina... nevies to z hlavy? :-)

Reply:
neviem, tez sa mi to nezda

Reply:
ono ta eventova cast je payload a ked to nejak dobre napises tu funkciu tak by to mala brat... takze potom tam mozes tie parametre pomenovat a neodkazujes sa na pole

Reply:
ok, mozem to opravit neskor